Index: lua-posix/lposix.c
===================================================================
--- lua-posix.orig/lposix.c	2013-06-29 17:11:42.885757128 +0200
+++ lua-posix/lposix.c	2013-06-29 17:11:42.881757129 +0200
@@ -11,6 +11,9 @@
 * With contributions by Roberto Ierusalimschy.
 * With documentation from Steve Donovan 2012
 */
+#ifdef __GNU__
+#define _GNU_SOURCE
+#endif
 
 #include <config.h>
 
@@ -673,7 +676,19 @@
 */
 static int Pgetcwd(lua_State *L)
 {
+#ifdef __GNU__
+	char *b = get_current_dir_name();
+	if (b != NULL) {
+		lua_pushstring(L, b);
+		return 1;
+	} else {
+		// we return the same error as below
+		return pusherror(L, ".");
+	}
+#else
 	long size = pathconf(".", _PC_PATH_MAX);
+	if (size == -1)
+		return pusherror(L, "pathconf");
 	void *ud;
 	lua_Alloc lalloc = lua_getallocf(L, &ud);
 	char *b, *ret;
@@ -686,6 +701,7 @@
 		lua_pushstring(L, b);
 	lalloc(ud, b, (size_t)size + 1, 0);
 	return (ret == NULL) ? pusherror(L, ".") : 1;
+#endif
 }
 
 /***
